### PR TITLE
ESASky: check whether response is valid

### PR DESCRIPTION
**Pull request by bsipocz**
This is to address partially #1168. Ideally this module could use the common `_download_file` method to deal with the downloads but that point beyond this PR.

Also includes removal of print statements in favour of the logger, and removing legacy string formatting.

I've checked this with Herschel data only, and it works sensibly.

### CODE REVIEW

File: astroquery/esasky/core.py [hunk 1]
```diff
@@ -373,10 +374,15 @@ def get_maps(self, query_table_list, missions=__ALL_STRING,
                             download_dir,
                             cache))
                     break
-        if (len(sanitized_query_table_list) > 0):
-            log.info("Maps available at %s" % os.path.abspath(download_dir))
+
+        if all([maps[mission].count(None) == len(maps[mission])
+                for mission in maps]):
+            log.info("No maps got downloaded, check errors above.")
+
```
Comment (imbasimba) [inline astroquery/esasky/core.py]: We could add this check to get_images also

File: astroquery/esasky/core.py [hunk 2]
```diff
@@ -547,19 +560,25 @@ def _get_maps_for_mission(self, maps_table, mission, download_dir, cache):
                         fits_file.close()
                         maps.append(fits.open(file_name))
 
-                print("[Done]")
-            print("Downloading of %s data complete." % mission)
+                if None in maps:
+                    log.error("Some downloads were unsuccessfull, please check "
```
Comment (keflavich) [inline astroquery/esasky/core.py]: -1 l

File: astroquery/esasky/core.py (hunk 1 above)
Comment (bsipocz) [reply astroquery/esasky/core.py]: Done

File: astroquery/esasky/core.py [hunk 3]
```diff
@@ -362,8 +363,8 @@ def get_maps(self, query_table_list, missions=__ALL_STRING,
             for mission in sanitized_missions:
```
[the remark this answers is not in the document]
Comment (imbasimba) [reply astroquery/esasky/core.py]: I believe it basically acts as a filter. So you only get the maps from missions you have specified with the 'missions' parameter, even though there may be more missions in your query_table_list

[the remark this answers is not in the document]
Comment (bsipocz) [reply astroquery/esasky/core.py]: hmm, OK. Even in that case there is no need for the loop I think, please review my last commit whether that makes sense.

**Pull request merged**